src/scripts/ibmcloud-auth.js

```diff
@@ -111,6 +127,40 @@ if (env.readerUsers) {
 	READER_EMAILS = env.readerUsers.split(',');
 }
 
+function ldapInit() {
+	var promise = new Promise((resolve, reject) => {
+		if (env.ldapServer && env.ldapPort && env.ldapBindUser && env.ldapBindPassword) {
+			var ldapUrl = `${env.ldapProtocol}://${env.ldapServer}:${env.ldapPort}`;
+			POWER_GROUPS = env.ldapPowerGroups.split(';');
+			READER_GROUPS = env.ldapReaderGroups.split(';');
+			if (bot) {
+				bot.logger.debug(`${TAG}: LDAP POWER_GROUPS=${POWER_GROUPS}`);
+				bot.logger.debug(`${TAG}: LDAP READER_GROUPS=${READER_GROUPS}`);
+				bot.logger.info(`${TAG}: Attemping to connect to LDAP at ${ldapUrl}.`);
+			}
+			ldapClient = ldap.createClient({
+				url: ldapUrl,
+				connectTimeout: 5000,
+				timeout: 10000
+			});
+			ldapClient.bind(env.ldapBindUser, env.ldapBindPassword, function(err) {
+				if (err) {
```

Comment:
My intent was to allow authorization initialization even if there was an error with LDAP as I want to make sure the auth middleware and NLC are in place so the bot is still locked down for usage.